app/controllers/devise/registrations_controller.rb

```diff
@@ -83,9 +83,10 @@ def cancel
   protected
 
   def update_needs_confirmation?(resource, previous)
-    resource.respond_to?(:pending_reconfirmation?) &&
-      resource.pending_reconfirmation? &&
-      previous != resource.unconfirmed_email
+    resource.respond_to?(:can_send_reconfirmation?) &&
```

Comment:
I can see this flash notice 
`You updated your account successfully, but we need to verify your new email address. Please check your email and follow the confirm link to confirm your new email address.`

So I did these changes as this should be displayed only when  toggle for reconfirmation is ON